store/v2/root/snapshot.go

```diff
@@ -0,0 +1,251 @@
+				if err != nil {
+					return sdkerrors.Wrap(err, "error at set the store schema key values")
+				}
+			}
+
+		case *storetypes.SnapshotItem_Store:
+			storeName := item.Store.GetName()
+			// checking the store schema is received or not
+			if !storeSchemaReceived {
+				return sdkerrors.Wrapf(sdkerrors.ErrLogic, "received store name before store schema %s", storeName)
+			}
+			// checking the store schema exists or not
+			if _, has := rs.schema[storeName]; !has {
+				return sdkerrors.Wrapf(sdkerrors.ErrLogic, "store is missing from schema %s", storeName)
+			}
+
+			// get the substore
+			subStore, err = rs.getSubstore(storeName)
+			if err != nil {
+				return sdkerrors.Wrap(err, fmt.Sprintf("error while getting the substore for key %s", storeName))
+			}
+
+		case *storetypes.SnapshotItem_KV:
+			if subStore == nil {
+				return sdkerrors.Wrap(sdkerrors.ErrLogic, "received KV Item before store item")
+			}
+			// update the key/value SMT.Store
+			subStore.Set(item.KV.Key, item.KV.Value)
+
+		default:
+			return sdkerrors.Wrapf(sdkerrors.ErrLogic, "unknown snapshot item %T", item)
+		}
+	}
+
+	// commit the all key/values to store
+	_, err = rs.commit(height)
+	if err != nil {
+		return sdkerrors.Wrap(err, fmt.Sprintf("error while commit the store at height %d", height))
+	}
+
+	return nil
+}
+
+// Snapshot implements snapshottypes.Snapshotter.
+func (rs *Store) Snapshot(height uint64, format uint32) (<-chan io.ReadCloser, error) {
+	if format != snapshottypes.CurrentFormat {
+		return nil, sdkerrors.Wrapf(snapshottypes.ErrUnknownFormat, "format %v", format)
+	}
+
+	if height == 0 {
+		return nil, sdkerrors.Wrap(sdkerrors.ErrLogic, "cannot snapshot height 0")
+	}
+	if height > uint64(rs.LastCommitID().Version) {
+		return nil, sdkerrors.Wrapf(sdkerrors.ErrLogic, "cannot snapshot future height %v", height)
+	}
+	versions, err := rs.stateDB.Versions()
+	if err != nil {
+		return nil, sdkerrors.Wrapf(err, "error while getting the snapshot versions at height %v", height)
+	}
+	if !versions.Exists(height) {
+		return nil, sdkerrors.Wrapf(sdkerrors.ErrNotFound, "cannot find snapshot at height %v", height)
+	}
+
+	// get the saved snapshot at height
+	vs, err := rs.getView(int64(height))
+	if err != nil {
+		return nil, sdkerrors.Wrap(err, fmt.Sprintf("error while get the version at height %d", height))
+	}
+
+	// Spawn goroutine to generate snapshot chunks and pass their io.ReadClosers through a channel
+	ch := make(chan io.ReadCloser)
+	go func() {
+		// Set up a stream pipeline to serialize snapshot nodes:
+		// Export KV Item -> delimited Protobuf -> zlib -> buffer -> chunkWriter -> chan io.ReadCloser
+		chunkWriter := snapshots.NewChunkWriter(ch, snapshotChunkSize)
+		defer chunkWriter.Close()
+		bufWriter := bufio.NewWriterSize(chunkWriter, snapshotBufferSize)
+		defer func() {
+			if err := bufWriter.Flush(); err != nil {
+				chunkWriter.CloseWithError(err)
+			}
+		}()
+		zWriter, err := zlib.NewWriterLevel(bufWriter, 7)
```

Comment:
Can we explicitly type what level of compression "7" specifies? Either in a comment or by defining a properly named constant variable for it.